packages/test/src/integration-tests.ts

```diff
@@ -1207,6 +1211,68 @@ export function runIntegrationTests(codec?: PayloadCodec): void {
     at stackTracer (test/src/workflows/stack-tracer.ts)`
       );
     });
+
+    test('Enhanced stack trace returns trace that makes sense', async (t) => {
+      const { client } = t.context;
+      const workflowId = uuid4();
+
+      const enhancedStack = await client.execute(workflows.enhancedStackTracer, {
+        taskQueue: 'test',
+        workflowId,
+      });
+
+      const stacks = enhancedStack.stacks.map((s) => ({
+        locations: s.locations.map((l) => ({
+          ...l,
+          ...(l.filePath ? { filePath: l.filePath.replace(path.resolve(__dirname, '../../../'), '') } : undefined),
+        })),
+      }));
+      t.is(enhancedStack.sdk.name, 'typescript');
+      t.is(enhancedStack.sdk.version, pkg.version); // Expect workflow and worker versions to match
+      t.deepEqual(stacks, [
+        {
+          locations: [
+            {
+              functionName: 'Function.all',
+            },
+            {
+              filePath: '/packages/test/src/workflows/stack-tracer.ts',
+              functionName: 'enhancedStackTracer',
+              line: 32,
+              column: 35,
+            },
+          ],
+        },
+        {
+          locations: [
+            {
+              filePath: '/packages/test/src/workflows/stack-tracer.ts',
+              functionName: 'enhancedStackTracer',
+              line: 32,
+              column: 35,
+            },
+          ],
+        },
+        {
+          locations: [
+            {
+              functionName: 'Promise.then',
+            },
+            {
+              filePath: '/packages/workflow/src/trigger.ts',
+              functionName: 'Trigger.then',
+              line: 47,
+              column: 24,
+            },
+          ],
+        },
+      ]);
+      const expectedSources = ['../src/workflows/stack-tracer.ts', '../../workflow/src/trigger.ts'].map((p) => [
```

Comment:
Having .. and ../.. in these files paths makes it more difficult to read and compare to the values in the trace above.

I'd suggest writing these as forward-only paths relative to the packages directory. Then, simply rewrite the `path.resolve`s below to something like `path.resolve(__dirname, '../..', p)`

Reply:
Don't think this is critical TBH